### ISSUE TITLE
Performance Issues

### ISSUE THREAD

**Issue by tom-doerr** [GothenburgBitFactory/taskwarrior#2201]
### Issue
Taskwarrior takes relatively long to output a report (0.8 seconds). 
0.8 seconds might not seem much, but I use shell scripts that call taskwarrior multiple times on each shell script call, e.g. to execute ```task +ACTIVE done``` before starting a new task, which builds up to a delay of around 4 seconds. 
Even if the performance would stay the same, which it won't, I would spend  16 hours a year waiting for tasks to start (at 15000 tasks a year, 4 seconds each).

### What I tried
- Disabling plugins => does not influence performance
- Removing the ```completed.data``` file => improves performance a lot, generating reports only takes 50 ms. Can I do that or will that brake something?

Is there any way to improve performance while keeping the completed tasks?

### Setup Info
Here are the file sizes of the relevant taskwarrior files:
```
-rw-rw-r-- 1 tom tom  20M Aug  6 17:27  backlog.data
-rw-rw-r-- 1 tom tom 3,4M Aug  6 17:10  completed.data
-rw-rw-r-- 1 tom tom  53K Aug  6 17:27  pending.data
-rw-rw-r-- 1 tom tom  30M Aug  6 17:27  undo.data

```

Here's my ```task diag``` output:
```task 2.5.1
   Platform: Linux

Compiler
    Version: 7.2.0
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++11

Build Features
      CMake: 3.9.5
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.5.8
 Build type: None

Configuration
       File: /home/tom/.taskrc (found), 1436 bytes, mode 100664
       Data: /home/tom/.task (found), dir, mode 40775
    Locking: Enabled
         GC: Enabled
     Server:
      Trust: strict
Certificate: , not readable, 0 bytes
        Key: , not readable, 0 bytes
    Ciphers: NORMAL
      Creds:

Hooks
     System: Enabled
   Location: /home/tom/.task/hooks
     Active: on-add.auto_tag.py    (executable) (symlink)
             on-modify.auto_tag.py (executable) (symlink)
             on-modify.timewarrior (executable)
   Inactive: modify-input.txt      (not executable) (unrecognized hook name)
             on-modify.save        (not executable)
             tags.config           (not executable) (symlink) (unrecognized hook name)

Tests
      $TERM: screen (356x74)
       Dups: Scanned 15415 tasks for duplicate UUIDs:
             Found duplicate 30fbcba9-277a-4406-8cd6-3629385f1bd8
             Found duplicate b72979b7-1a69-4188-b1c3-09ecdb5cd612
             Found duplicate 789f030c-69e6-4538-a1d9-8f50445242c4
             Found duplicate 59b085f6-8b1e-4de2-85f7-279b98113d89
             Found duplicate c5f9f3fa-4073-4266-8b17-0d06cf694dc3
             Found duplicate e54a5e9b-4485-4116-9e81-8468adb05cda
             Found duplicate 335922b1-e4a0-4d39-b9d8-ca628a2335ba
             Found duplicate 125bbe74-8f11-4166-8c86-3746e8597037
             Found duplicate ff633b78-3109-46ea-8012-fc874b87f12b
             Found duplicate f8df86cd-839a-4df4-8989-566a29048609
             Found duplicate d7660bde-0f31-4c2b-8206-96cd491a3f9e
             Found duplicate 81026ab4-7c0c-47bb-b609-c67481cec8ff
             Found duplicate 318981f1-41e9-4372-ba3e-a14c368d8d00
             Found duplicate 672d319f-1216-4abc-98d9-3a9edcd2c94d
             Found duplicate 20415832-92a3-46cc-872d-7245ae5863ac
             Found duplicate 1ea15e7e-335f-427e-a77b-2e7e8521a741
             Found duplicate 377e100d-b164-4dc6-a28b-7ee32f70e9ba
             Found duplicate 6d3af0a1-9f29-486b-8c0f-8037c2cf6bcc
             Found duplicate 56171909-36d9-485f-8285-fff65a608736
             Found duplicate 29b01f47-7fd1-4a22-b34f-3d9b94cc19a4
             Found duplicate 2c5fa8c3-f7c6-4f66-ad28-21ea74457d98
             Found duplicate 40e0bf98-03a5-40ce-bf31-e73e0334fcc0
             Found duplicate 1fee270d-2b25-4708-9373-3eb09d6fc29e
             Found duplicate a98f5730-1dbb-424b-ade8-ea768a9f1dab
             Found duplicate 12d59f0c-ea53-4246-93a7-8185b0bbf8ce
             Found duplicate e10d8a40-86d1-4346-a481-2b0710d66cb5
             Found duplicate c42db3a4-163f-41e0-bd53-7b6889ea5e70
             Found duplicate 24af6d1d-67fb-484b-80c9-08c732582fa3
             Found duplicate ddce2516-22be-4e6f-9c5b-cfded0f651ab
             Found duplicate 9aef008b-074a-45a5-bfa2-57c6e8e0a802
             Found duplicate 38bd1aee-63b2-44ba-9a12-4c2a8c576dd9
             Found duplicate 54b515d2-ef53-4b5f-82dc-ebcd70ebb554
             Found duplicate f2ee3f63-67cb-4f87-917e-0085139649b8
             Found duplicate e6dd8342-7321-4b0f-8877-35f866d5ef67
             Found duplicate 70ba74b5-2881-44ba-bb27-5365ed98103f
             Found duplicate d070bdce-3f78-4515-8400-d334d6aac77d
             Found duplicate c9d39b8f-2118-4230-8e5d-3c8d97b860ee
             Found duplicate 8576682d-dc32-40fc-9796-fe855ecf21c1
             Found duplicate 09d1c745-a17f-439c-9f10-8b8d6e29c3b4
             Found duplicate e41da72e-e82c-4476-9377-69959c568235
             Found duplicate 9e77ce92-d8ed-4b5d-9076-c6455dcf353a
             Found duplicate e8671c82-8593-4000-af53-efd35e60414e
             Found duplicate b47525d3-9a55-4ca1-826e-d0d3f3d7fa8d
             Found duplicate c19624c9-f619-4c71-bc49-900b82cb07c7
             Found duplicate c19624c9-f619-4c71-bc49-900b82cb07c7
             Found duplicate 76a74347-1bd3-431d-a5a9-d23b4eb38778
             Found duplicate 1ffd7f5e-3710-42b9-bd4b-5ef25a4c01e3
             Found duplicate 98c8dda3-f520-43ed-84fd-14203ff348b1
             Found duplicate c998292d-9748-4fa1-8431-8c0413750e9f
             Found duplicate a86dc05c-b576-4721-a092-a20729c9e406
             Found duplicate 55b09f8e-0467-4f30-b1d8-955345e2ced5
             Found duplicate 705ba075-4c0c-4547-b9f4-f2e131409401
             Found duplicate fe0f3c26-2376-4c0e-8409-9fd15a0a5bf4
             Found duplicate c5330efe-5029-4763-9ffa-05f12c61e6f7
             Found duplicate 66331839-d1f9-44f1-9502-f8208dfcc617
             Found duplicate 22b7af9e-5a09-41e7-ba87-f56207d3f870
             Found duplicate 37572995-7b68-4af5-ab29-8fca44485a4c
             Found duplicate 07212b43-be8e-4c27-ac14-8be9cca8ec96
             Found duplicate 39f7ad39-ea8f-4f89-b3b8-c0944d1555bb
             Found duplicate ee3e4843-92db-4d53-accf-536bed849340
             Found duplicate aae4e911-d619-410a-9884-ba4cc1c60696
             Found duplicate 71b84a58-e017-446e-9b60-49c36c2a65bf
             Found duplicate 6df223e1-6512-42ae-8ddb-a6b8b3400375
             Found duplicate dab7763a-5ff2-47cc-b3a6-6dab061a4930
             Found duplicate 1b03770c-f2bb-4045-81d0-0cb152350eea
             Found duplicate b789d4c4-5e19-4487-8f79-d79cbbb1503f
             Found duplicate 5b451375-cd8c-4db1-96d2-95b76886efc6
             Found duplicate 3bd55399-26df-4691-8a4d-091ee3ab8548
             Found duplicate 4a4797ee-7188-4188-8091-e84d7a559842
             Found duplicate e43ccd1f-6b8a-4e1f-a3fc-bcd137fc4ad4
             Found duplicate 37f0f5a0-3afe-48ae-9235-a7ef603e455c
             Found duplicate 7dc53215-2a28-4254-bb0d-f763dd3874e0
             Found duplicate a90d564e-4b4f-417d-be65-903be9678abe
             Found duplicate 6d68435b-b1a1-45dc-a842-395326f8255e
             Found duplicate 0c1cfe6b-e7a3-42f4-8801-aa1bc885bcc0
             Found duplicate 70dab55f-f0bd-49ca-857e-d5e6d30ed535
             Found duplicate 5b637ad8-2d4b-4322-9616-1eee2e863e9b
             Found duplicate ef44c41e-be71-4e33-8bc3-7cd9cbd3e0cb
             Found duplicate 4898a51a-18de-4e81-b50f-22e3f68f62fa
             Found duplicate a586e29e-e348-45ce-8b46-8fe28cee2694
             Found duplicate 32c4b1c9-efbc-4d86-b634-0be44d1849fd
             Found duplicate 917b1882-d63f-4c46-ad1a-711bd02f6bdb
             Found duplicate ec56630f-1320-44fd-91b4-ff6c802d96bf
             Found duplicate db5a151d-da00-4306-897f-f8232199647a
             Found duplicate 9627e7bb-f639-4de8-b425-5696de81e9f3
             Found duplicate 18dd4d3e-0190-4773-937e-ad82b890b60c
             Found duplicate e6f80510-71b2-4b43-8000-ad4e8bb684b9
             Found duplicate 4ddb7b1a-3da5-449c-8b3f-ca24507d944c
             Found duplicate c64a57b5-f1bc-46e9-a501-5af14c01ab93
             Found duplicate 5c6c8a08-9c7d-47a1-b7d8-bdd603efe11e
             Found duplicate 6489d477-bcd9-4624-834d-6759c50df427
             Found duplicate 4f4c7d53-95a3-4544-99e3-e765a0fb89a2
             Found duplicate 68b491d9-41e6-4070-8bf4-d7f24572367a
             Found duplicate 5beb341b-6530-435f-86dd-e4c7b76c135a
             Found duplicate 8ab5d91d-75ff-47f6-8945-83d008e92e46
             Found duplicate ad4bc642-d940-4001-9bb1-9de32a3c7112
             Found duplicate 896650fb-03b4-4260-94dd-f245b06465b8
             Found duplicate c1552471-8f7b-4d4f-a5be-ed0f09285168
             Found duplicate 40a4e942-1dbe-4b2a-80d0-56115a5770f4
             Found duplicate c6311592-e381-43d6-a0c2-d4d36c4eb369
             Found duplicate bb74c520-f4d5-4d1d-bd05-2c37e90ef876
             Found duplicate fa1186a9-b5bd-429f-9990-95b8abaaea26
             Found duplicate a37590f0-7e04-4d3a-a592-5ac3631e413b
             Found duplicate f5996c0f-f0ba-439c-b804-f9db7f2f0b50
             Found duplicate 6c0ff4b5-850d-49ad-b321-2e5785edea5e
             Found duplicate 99d7af9b-0414-4ba0-a7fd-4d603a5b2650
             Found duplicate b1ca0751-b81c-4282-9466-067607081f91
             Found duplicate 71fb33d2-fc2f-4759-baaf-163f6cfa695e
             Found duplicate 3b0b9780-3d09-46e4-a561-70f607c275b7
             Found duplicate 3a3b6e08-3fac-45b4-93b6-2e68c60456bd
             Found duplicate a60b1d00-a69f-4408-98c9-43b668fca544
             Found duplicate afaa99cb-3fd7-4689-8034-384aaa415101
             Found duplicate 33e57539-e62a-468f-93e9-5ba0abd71872
             Found duplicate 0c26c31c-ff6a-4d9f-8bd4-0a499e18c6ba
             Found duplicate 151a4e81-e8f8-444a-a216-0555e4b8bb84
             Found duplicate 053979dc-aca5-4b12-bc01-3d87ae86c306
             Found duplicate fe5fcac0-c65e-4793-b001-f41df3f17226
             Found duplicate e750d119-b6aa-4424-b750-8e8028830027
             Found duplicate a2d9ec0a-07a2-44c2-8b9c-cd309d109c85
             Found duplicate 4ef03b35-1e48-46ff-bfa7-6a224eaf7d0a
             Found duplicate 7704ba8e-dd60-47ff-8628-f1457c9bf875
             Found duplicate 2731a670-e59b-459a-97f9-94ae9a63128a
             Found duplicate c9b945be-068b-439a-94d6-50cbc7944b48
             Found duplicate 01ff0932-20b8-477f-98e7-fae967a945a4
             Found duplicate 15fdb001-0b41-4b4d-af51-2c4a8f495b49
             Found duplicate 7c1a1a6e-0730-48f9-bf33-3c444b64430b
             Found duplicate b1a443b0-c94a-42cf-9f4c-acf10ac2da2f
             Found duplicate 20c389ec-07de-486d-b52b-54197f5bd81e
             Found duplicate b372716b-20f7-4b64-827d-127a21ee8e9f
             Found duplicate 3a651e36-c1c1-4c7d-842e-3b181eed9369
             Found duplicate 1472360b-0b61-49d0-ac86-ec69b41e5f7b
             Found duplicate b4edfb06-7f7d-4cec-a819-9ed288b11c6c
             Found duplicate 45e8f5fc-ba41-4f2b-adec-54e39124c4b9
             Found duplicate d980fbb9-b51d-42a9-8b42-264b85acad8e
             Found duplicate 0ca9dbab-c10f-4db6-9a8e-9a5dc06bc430
             Found duplicate 72d27409-91d1-4699-a37b-3ae95c15267a
             Found duplicate c6d542e8-3601-4672-9843-c0dad1dd6e03
             Found duplicate 6a176a50-2a37-4963-ac6a-5636d3843ad3
             Found duplicate a1dec7e7-846c-42c5-a1bc-477a55894230
             Found duplicate 650bc129-39ab-4ffc-a01f-818b6751b7c2
             Found duplicate 12a2ae35-6442-476e-a8b3-c41b4881763f
             Found duplicate 0ae7b6e1-95c0-4484-8bf6-cbc0d04d3765
             Found duplicate 017b7f3f-da69-4ae3-a4c1-a7e0b5725938
             Found duplicate 1ffe91ea-746a-4bd9-80e7-02dc87f03026
             Found duplicate f0a37ad5-3e6b-4049-ad2c-fe27ccf0f41d
             Found duplicate ff80ba36-b476-4828-b279-8bd90ef5a913
             Found duplicate 06e5840b-d7f3-46c8-ab0c-fb18c7cb9ad5
             Found duplicate ac0f5f42-0001-450f-a107-b2c850e79fd5
             Found duplicate a59b0d9b-49c0-4818-a8a8-768b5dd30571
             Found duplicate b0308cee-5eed-4fe9-b7fe-76654521927c
             Found duplicate b2d6b08f-0f07-4247-83f7-54439e5a95a7
             Found duplicate 2eb1f89e-c5ac-4d9a-af55-d27016379e49
             Found duplicate 819fb11f-7dee-4929-9a28-73c5fa257162
             Found duplicate 7ef6ebf2-6402-422e-add7-2638956ae6b6
             Found duplicate dad47022-d322-43b4-90d8-a3814487a38a
             Found duplicate 41fa45f6-e405-4bdf-9964-b3386a75988d
             Found duplicate c012b109-de61-4022-ade1-77825926e6fe
             Found duplicate 49abbfb9-c64b-48bb-8b55-86af93c8ff9b
             Found duplicate 8677e256-98af-4710-a654-f7cb46642c62
             Found duplicate 2269012f-8617-4c5c-bafd-93e59bf24f70
             Found duplicate b3ed1e0b-182b-44be-8c9a-ea4cbc6e3709
             Found duplicate 1a4ba855-e8e0-4a48-8238-820e412630e4
             Found duplicate 25bfa06d-a192-4d5b-916e-2cd60df3d190
             Found duplicate 8dd52ece-ef81-4915-902e-5ef6ea8aa0ae
             Found duplicate 0f5fc51e-bcef-47fd-bc52-5463a7fd78ee
             Found duplicate ff2e2551-6d20-4f04-a5f8-540a3efeed76
             Found duplicate bc45cb17-f5e0-46ae-a78d-353a3847ffc1
             Found duplicate 63ab99ce-85fc-46ff-aa94-1420185d920c
             Found duplicate ab3923b8-18d0-4902-9672-89e867a68fcc
             Found duplicate a655198d-976a-4711-a174-8bec356b25d3
             Found duplicate 6e395796-b1ad-47c2-8d99-5d5d60da878f
             Found duplicate d4103ed1-45d4-44d2-a465-a665528ff13a
             Found duplicate 40eb226d-4b63-4b67-84de-8197203ce64d
             Found duplicate c19a5df6-2cb0-4e7d-b1c4-93b960ce7e7c
             Found duplicate 7c8dd2ef-e9a7-4a7e-98b3-3e301f6098d4
             Found duplicate 1959ac10-072f-4275-8c4d-f2b5806d04ca
             Found duplicate 5fba4c91-55e3-4812-a33e-868f32706d52
             Found duplicate 57f5fc6c-bc39-4fb7-be61-7096a4911845
             Found duplicate a8cd71d9-6b1a-4774-b6c2-c6efd4cff1b7
             Found duplicate c978d5f6-2c03-49c4-9bce-a45f365e6154
             Found duplicate c20a42d8-213e-4cf0-9f93-62dfadc9758e
             Found duplicate df5c4cd3-9bff-4244-ab53-3d2d8eb2c07e
             Found duplicate df5c4cd3-9bff-4244-ab53-3d2d8eb2c07e
             Found duplicate de2200e7-9430-4f20-9080-c4c73aaad61d
             Found duplicate c74ea950-3308-44e7-889e-248ce4eb0241
             Found duplicate 672f9f7a-83a8-4e13-b131-8768bc8e0d8c
             Found duplicate c0fece4a-ce9b-4945-8ae9-3863d27493e1
             Found duplicate d689e6b7-d46f-4027-8116-fb45fd375a3b
             Found duplicate ac70ce5a-32ba-4174-a96c-848793564f5e
             Found duplicate d8e0792d-50e1-4d46-853d-5948c8b87c75
             Found duplicate 74b96a6e-a449-41c6-82c1-fd567c6d39b3
             Found duplicate fab66112-12c6-4f50-85bd-cab10baa387e
             Found duplicate aa909b9c-d017-4821-8ada-355c2b387748
             Found duplicate 303c2263-f884-4b3a-8e45-3c7c430b8b84
             Found duplicate 0144d30d-c38a-479f-b555-e5b504d33ac6
             Found duplicate e5f8b170-11c8-4468-93d1-2ab61baf3745
             Found duplicate 778c6ce4-29e7-4903-97d8-4427df82e09d
             Found duplicate aa33052a-c2e3-4abf-8623-045493627b7f
             Found duplicate f9cef0e5-fcf3-464c-acc2-6dd95240ac38
             Found duplicate e011d91a-76c6-45b8-9eab-7c405fb4b69e
             Found duplicate 578a0241-3714-4758-aed5-508fdc6d67c0
             Found duplicate d7c9ec2a-22b6-496b-aa09-acdfc35687dd
             Found duplicate e2318c27-f66c-4540-b915-08b636962e95
             Found duplicate 9ffaf0c0-c789-419e-84ff-eb3a6de5d56f
             Found duplicate ba7bfc72-94ec-432e-a069-e109da19e696
             Found duplicate dd92e58f-7fe0-4054-8667-2e48fd8da4c5
             Found duplicate a79f28ea-30e9-45d0-a2df-4612b0561b34
             Found duplicate 73ab8850-9b63-4561-ba0c-4ba5dbb7db1a
             Found duplicate 7022ee8d-6ef4-4407-b96d-6493e340ee65
             Found duplicate 68d6287c-dc4f-4bb4-84a5-ad2a390a1667
             Found duplicate 30b00be8-f268-4187-afeb-2ef42df2c63f
             Found duplicate 33cbe54c-da56-49c1-a05c-115dd0fcfbbc
             Found duplicate cde03066-4f38-4771-b53c-996c9c7f50e7
             Found duplicate 692d2698-9cef-4c1a-a54d-1e50732b00ee
             Found duplicate f75ce7cf-1720-47ce-a868-c12f3dc49367
             Found duplicate 4f5f3cc9-e75a-4f56-8b87-a3d001971c86
             Found duplicate 718d0939-68ae-4a4e-9b87-7d62910ad119
             Found duplicate 821f9e69-28b1-4e4e-b242-54a3d5eed1c6
             Found duplicate e15c4a77-0e6a-43e5-9b6b-35f1e7de5ed2
             Found duplicate 63318d20-b236-42b8-8b1e-44b01dc3dda0
             Found duplicate fa573555-5e5b-4ec4-aa8c-04fe21481aac
             Found duplicate ebca5366-9c4e-4e3c-ab1a-a62e7815550c
             Found duplicate d9b89cf0-4fb6-48eb-9dc7-5a4db47122be
             Found duplicate b3ec5b7b-0473-4d4c-9613-cc3911602f36
             Found duplicate 57d715ae-aa59-4485-8604-aa680a1c467a
             Found duplicate 877d9ce5-f899-48ed-9959-db14cb97b786
             Found duplicate 3187bedb-a81c-4141-bf3b-de04af8b2045
             Found duplicate f4a02c1d-0254-4c8d-ac3e-b4c0fa298562
             Found duplicate b26bd561-191f-4c48-9c9e-ff3611fa3e87
             Found duplicate d2659117-3925-44b9-a51a-825ea143667a
             Found duplicate a270d620-8f2e-4dd0-8692-efd66c2d877a
             Found duplicate 38440180-9fdb-44ed-964b-49f2c9f56d32
             Found duplicate 359b0c9b-d0a2-41d8-8c53-cfbf253ad365
             Found duplicate 083d4a0a-c600-4dd6-b81a-dadaf917a7e6
             Found duplicate ecdd0a77-5940-4a6b-ba2d-1563c59435b7
             Found duplicate 2d35f5b7-8f12-4453-9f89-ff5810a7226e
             Found duplicate c2e6fa46-a580-426d-a59e-888f78d3cbf3
             Found duplicate 76534d43-6581-49f8-8861-20811920b6b7
             Found duplicate b3a5717b-d5f7-40be-adbb-c1f55dc5443d
             Found duplicate c3e68ace-c83e-4e59-a0ea-43aeab7581e3
             Found duplicate 9de43858-1d88-4761-8d38-234ac1f0f765
             Found duplicate 790f4c19-0c62-4d5f-9b9b-502fe534f5cc
             Found duplicate 91d9978c-1f44-486d-b969-ee3118f77185
             Found duplicate 6ef1e913-3f0f-4c83-8bf5-7dff73818717
             Found duplicate 9c9be8c0-28f4-4a1e-a55a-30c78224aa3b
             Found duplicate 6b357f9e-1cc0-4c10-a133-449274e47ac3
             Found duplicate 4fd8c1ee-5935-43aa-a4a9-939e9e79fab1
             Found duplicate 7ef0a24e-d56f-472c-a18a-ed1e3205fd36
             Found duplicate 3583d1d8-7381-4734-b730-3a179682760c
             Found duplicate b2f236d0-f0ac-4f9f-aa70-5df3294dc5fe
             Found duplicate bf67d44d-e4e7-4d90-978e-4cf61edb75e7
             Found duplicate 7c72b089-ad60-4ef7-bd76-afc59dcdeec8
             Found duplicate b4a98300-79ab-4f01-bf83-28613247e8af
             Found duplicate 6b79ec04-1d55-4f7a-9b6c-ad0a39c606b0
             Found duplicate d1f1c63f-e46b-425f-9e8d-15f0b610a790
             Found duplicate fdbe2546-72c9-4499-a237-c522f25753d5
             Found duplicate edd8b8ea-17e4-49d2-aca3-32bb73b7da33
             Found duplicate c867ed2e-7cb2-49ac-81ff-c2995c4ae5a2
             Found duplicate 09e725ba-0b3c-45b3-b3ee-3590a8540759
             Found duplicate 66e75f17-962f-494a-8952-735544c28e98
             Found duplicate 76f6b39f-9085-469f-973f-80022037a94a
             Found duplicate 0a4dc3bf-5267-41c7-9717-c0bbfe1d8447
             Found duplicate db98044b-b285-40aa-a7a2-4ae659872087
             Found duplicate 9ed83266-30a9-45ef-bee1-9dc820a05ed4
             Found duplicate c9926706-a4cd-43fa-b8f3-d60bd4726703
             Found duplicate 78136bc9-aaf9-4576-aa67-17ae0599f930
             Found duplicate bc9a474f-0679-4b90-9de5-2027d9e95fda
             Found duplicate bd1679fb-88e4-4734-86de-ce6f07d40441
             Found duplicate f5867bf4-6f5a-4be4-ae3c-4206c17f9c0b
             Found duplicate d52a2bda-ddf6-45ef-a88f-e3c78d23df4b
             Found duplicate d9d71831-8003-41bc-a33a-044f7bc6d853
             Found duplicate 64dc8fb0-01e3-42b5-8bdc-e727a229c305
             Found duplicate 8858d5e9-f615-4d93-b21f-aae64bba90ba
             Found duplicate caf3df3d-baa3-40d9-9f8b-6a08d291d024
             Found duplicate f52a8187-2d3c-42f9-bf85-64c05b21d568
             Found duplicate b1d4f015-579a-4e04-afa5-6051768bc215
             Found duplicate 2291f19b-9ef7-4df9-b1e8-8c2745809535
             Found duplicate 5374fa73-28d1-411b-aa4a-cd20c7085679
             Found duplicate 3782116e-f812-419e-8909-5158fd13947c
             Found duplicate 18552d76-3003-4f11-a011-f9389ecb6eaf
             Found duplicate d7a2a135-548f-4243-bd32-09c58082dbc7
             Found duplicate a6cdd43f-31ec-4748-9635-9d3991906c0d
             Found duplicate 9dd560b1-ec35-4c9f-9b4c-304c3fdae07f
             Found duplicate 9dcef9a3-38f1-4758-9a1c-62d7f9de25fc
             Found duplicate 7c67f687-0a3d-4403-9c93-7a878d792f2a
             Found duplicate 2cf36ddd-f257-482a-ac77-61bb6a00b400
             Found duplicate c66629ea-6227-4647-8ffc-97468fa76d5e
             Found duplicate 1d3844f9-c52b-40be-884d-c04a8d7b5c10
             Found duplicate cdd415ee-2514-4a6f-ab6f-669f75ea146c
             Found duplicate b2f864a3-0b6e-4e75-af80-cef11bcf2710
             Found duplicate 79c4e082-62cf-4b9a-a321-1ea9eb5d1858
             Found duplicate 183f2356-2f72-497e-a2d0-d474aed6266d
             Found duplicate b07e4140-d030-4ad9-aeae-3a8c5a81f230
             Found duplicate 4f690283-79d3-4cea-80ff-4fca75c1c8aa
             Found duplicate 653d9170-4802-44b3-b305-ef19cb869997
             Found duplicate 6f7472cc-b428-430e-8066-76ca69185fd8
             Found duplicate befed424-210c-4d9c-83eb-09e7a9a81bce
             Found duplicate 2f9c136c-26b5-48f4-8f0b-292c37185ff7
             Found duplicate 29cfabe0-b13d-4e07-a8dc-2bef85b2f6dc
             Found duplicate 0aa7ad5b-a943-43be-8f96-d5436c87c5bc
             Found duplicate ec3e6c55-4f4c-4832-81e9-2d5d149fbfcf
             Found duplicate 5a43ef55-28ce-4fcb-b8fd-aeeb3eccbeef
             Found duplicate ddc12c5d-8473-4b17-9590-a06c735c103a
             Found duplicate 177c3603-0685-4d21-9eed-edc5c30fa079
             Found duplicate 5d7ffd32-c48e-4c5b-98e5-a9feef7be8c0
             Found duplicate 3cc3f070-0ed8-47f1-ae0d-a65c7adf7edc
             Found duplicate 3de1565b-5b56-4c0f-8bd5-235c26f8a8c7
             Found duplicate d4215321-895b-483f-9bcf-649ece44faa8
             Found duplicate ab7ce5fa-78ce-450f-9628-43ce71965ca5
 Broken ref: Scanned 15415 tasks for broken references:
             No broken references found

```

**Comment by pbeckingham** [comment]
There are some things here to consider about performance:

- You can delete completed.data. It contains completed and deleted task data. If you don't need that data, and can tolerate missing features such as burndown reports, project statistics, and the ability to resurrect old tasks, then you won't miss it.
- You don't use Taskserver, so you can delete backlog.data.
- If you have no plans to use the undo command, you can delete undo.data.
- Taskwarrior doesn't read those large files for the fun of it, it only reads them when it determines that it is necessary.  If you can restrict your command to only apply to pending tasks, the other files may not be read.

**Comment by tom-doerr** [comment]
Thanks for clarifying! Then I will just move those large files.
Regarding your last tip of restricting my commands to the pending tasks:
`task +PENDING` actually takes slightly longer than `task` does.

**Comment by pbeckingham** [comment]
Adding terms to a filter will always slow it down. The important thing is that certain filters do not need the completed.data file.

Look at the src/Filter.cpp pendingOnly() method.  It is responsible for determining if Taskwarrior can take the shortcut of not loading completed.data. It could use enhancement.

**Comment by tbabej** [comment]
@tom-doerr I addressed the most urgent performance issues in the code, the common filters now run 2-8x faster depending on the query. Additionally, I made sure +PENDING / +ACTIVE tags circumvent loading `completed.data`, as @pbeckingham suggested above.

That's the reason why `task +PENDING` did not really help in this case, `task status:pending` would though. In any case, this should hopefully would be decent enough of a speedup, feel free to reopen if the performance problems persist.

Ideally in the future we can setup a more involved performance measurement tracking to ensure that the performance is maintained even if new features are added.

**Comment by tom-doerr** [comment]
Thank you @tbabej !
